pandas/core/algorithms.py

```diff
@@ -981,14 +981,60 @@ def select_n_frame(frame, columns, n, method, keep):
     -------
     nordered : DataFrame
     """
-    from pandas.core.series import Series
+    from pandas import Int64Index
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
```

Comment:
```
def _is_valid_dtype_n_method(dtype):
    return ((is_numeric_dtype(dtype) and not is_complex_dtype(dtype)) or
                needs_i8_conversion(dtype))
```
then you can have separate messages for a series calling this (alone) and a dataframe with many columns.